### PR TITLE
Add stubs for `TgCrypto`

### PR DESCRIPTION
**Pull request by ViZiD**
Stubs for https://github.com/pyrogram/tgcrypto

### CODE REVIEW

File: stubs/TgCrypto/tgcrypto/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,6 @@
+def ige256_encrypt(data: bytes, key: bytes, iv: bytes) -> bytes: ...
+def ige256_decrypt(data: bytes, key: bytes, iv: bytes) -> bytes: ...
+def ctr256_encrypt(data: bytes, key: bytes, iv: bytes, state: bytes) -> bytes: ...
+def ctr256_decrypt(data: bytes, key: bytes, iv: bytes, state: bytes) -> bytes: ...
+def cbc256_encrypt(data: bytes, key: bytes, iv: bytes) -> bytes: ...
+def cbc256_decrypt(data: bytes, key: bytes, iv: bytes) -> bytes: ...
```
Comment (srittau) [inline stubs/TgCrypto/tgcrypto/__init__.pyi:6]: Thanks!

These functions use `PyArg_ParseTuple`, so they only accept positional-only arguments. Also, using `y*` means that they accept any buffer object:

```suggestion
from typing_extensions import Buffer

def ige256_encrypt(data: Buffer, key: Buffer, iv: Buffer, /) -> bytes: ...
def ige256_decrypt(data: Buffer, key: Buffer, iv: Buffer, /) -> bytes: ...
def ctr256_encrypt(data: Buffer, key: Buffer, iv: Buffer, state: Buffer, /) -> bytes: ...
def ctr256_decrypt(data: Buffer, key: Buffer, iv: Buffer, state: Buffer, /) -> bytes: ...
def cbc256_encrypt(data: Buffer, key: Buffer, iv: Buffer, /) -> bytes: ...
def cbc256_decrypt(data: Buffer, key: Buffer, iv: Buffer, /) -> bytes: ...
```

**Review by srittau — CHANGES_REQUESTED** [review]

File: pyrightconfig.stricter.json [hunk 2]
```diff
@@ -94,6 +94,7 @@
         "stubs/setuptools/setuptools",
         "stubs/shapely",
         "stubs/tensorflow",
+        "stubs/TgCrypto",
```
Comment (AlexWaygood) [inline pyrightconfig.stricter.json:97]: It seems like the type annotations for this package are complete, so this exclusion isn't necessary; we can just use the stricter pyright settings on this package:

```suggestion
```

File: stubs/TgCrypto/@tests/test_cases/check_tgcrypto.py
Comment (AlexWaygood) [inline stubs/TgCrypto/@tests/test_cases/check_tgcrypto.py:1]: Thanks for writing tests for these stubs! However, I think for htis pacakge, they're probably not necessary. The annotations for this package are very simple, and adding more tests for such simple stubs slows down our CI unnecessarily, in my opinion. I really appreciate you taking the time, but I think we can probably do without tests here.

**Review by srittau — APPROVED** [review]

**Comment by srittau** [comment]
(Not sure why stubtest had its knickers in a twist here, but thanks for adding the allowlist.)

File: stubs/TgCrypto/@tests/stubtest_allowlist.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+tgcrypto.cbc256_decrypt
```
Comment (srittau) [inline stubs/TgCrypto/@tests/stubtest_allowlist.txt:1]:
```suggestion
# stubtest doesn't recognize these function as taking only positional-only arguments.
tgcrypto.cbc256_decrypt
```

**Review by srittau — COMMENTED** [review]
One suggestion for documentation purposes.

**Pull request merged**